src/Hcal/HcalDigiProducer.cxx

```diff
@@ -1,392 +1,235 @@
-    hcalXpos[detIDraw] = hcalXpos[detIDraw] / hcaldetIDEdep[detIDraw];
-    hcalYpos[detIDraw] = hcalYpos[detIDraw] / hcaldetIDEdep[detIDraw];
-    hcalZpos[detIDraw] = hcalZpos[detIDraw] / hcaldetIDEdep[detIDraw];
-    double meanPE = depEnergy / mev_per_mip_ * pe_per_mip_;
-
-    ldmx::HcalID curDetId(detIDraw);
-
-    int cur_subsection = curDetId.section();
-    int cur_layer = curDetId.layer();
-    int cur_strip = curDetId.strip();
-
-    if (curDetId.getSection() == ldmx::HcalID::BACK)
-      numSigHits_back++;
-    else if (curDetId.getSection() == ldmx::HcalID::TOP ||
-             curDetId.getSection() == ldmx::HcalID::BOTTOM)
-      numSigHits_side_tb++;
-    else if (curDetId.getSection() == ldmx::HcalID::LEFT ||
-             curDetId.getSection() == ldmx::HcalID::RIGHT)
-      numSigHits_side_lr++;
-    else
-      std::cout
-          << "WARNING [HcalDigiProducer::produce]: HcalSection is not known"
-          << std::endl;
-
-    // need to add in a weighting factor eventually, so keep it that way to make
-    // sure we don't forget about it
-    double energy = depEnergy;
-
-    // quantize/smear the position
-    float cur_xpos(hcalXpos[detIDraw]), cur_ypos(hcalYpos[detIDraw]),
-        cur_zpos(hcalZpos[detIDraw]);
-
-    // for back HCal, get PEs with attentuation
-    if (cur_subsection == 0) {
-      float distance_along_bar =
-          (cur_layer % 2) ? fabs(cur_xpos) : fabs(cur_ypos);
-
-      // increase the PE count to the case with no attentuation (assuming 80%
-      // attenuation on the pe_per_mip number @ 1m)
-      meanPE *= exp(1. / strip_attenuation_length_);
-
-      float meanPE_close =
-          meanPE * exp(-1. * ((half_total_width - distance_along_bar) / 1000.) /
-                       strip_attenuation_length_);
-      float meanPE_far =
-          meanPE * exp(-1. * ((half_total_width + distance_along_bar) / 1000.) /
-                       strip_attenuation_length_);
-      float PE_close = random_->Poisson(meanPE_close + meanNoise_);
-      float PE_far = random_->Poisson(meanPE_far + meanNoise_);
-      hcalLayerPEs[detIDraw] = PE_close + PE_far;
-      hcalLayerMinPEs[detIDraw] = std::min(PE_close, PE_far);
-
-      if (cur_layer % 2 == 0) {  // even layers, vertical
-        cur_xpos =
-            (super_strip_width * (float(cur_strip) + 0.5)) - half_total_width;
-        cur_ypos =
-            hcalYpos[detIDraw] + random_->Gaus(0., strip_position_resolution_);
-      }
-      if (cur_layer % 2 == 1) {  // odd layers, horizontal
-        cur_ypos =
-            (super_strip_width * (float(cur_strip) + 0.5)) - half_total_width;
-        cur_xpos =
-            hcalXpos[detIDraw] + random_->Gaus(0., strip_position_resolution_);
+      std::vector<ldmx::HgcrocDigiCollection::Sample> digiToAdd;
+      if (hgcroc_->digitize(hitID, voltages, times, digiToAdd)) {
+        ldmx::HcalDigiID digiID(section, layer, strip, 0);
+        hcalDigis.addDigi(digiID.raw(), digiToAdd);
```

Comment:
This is where you would decide which end the single-ended readout would read-out on. I don't think think this is something can be calculated; I think you just have to _define_ which end is read-out for each section. For example, the RIGHT hcal section is probably being read out from the negative end because the positive end is up against the TOP hcal section.

Reply:
I implemented that [here](https://github.com/LDMX-Software/Hcal/blob/909d808e9e6d29c9336b6fcf57a2c4e868e0ee0d/src/Hcal/HcalDigiProducer.cxx#L220-L239). But reading your comment in more detail, yes the TOP hcal section will read out from the positive end  (0)  (this is positive x)  and the BOTTOM hcal section from the negative end (1), both have bars oriented in x right? Whereas the LEFT hcal section will read from the positive end (0) (this is positive y) and the RIGHT hcal section will read from the negative end (1) (this is negative y). Let me know if this logic makes sense, just confused on why you related the TOP and RIGHT hcal section in your comment.

Reply:
Just wanted to say that I went over the implementation and documented a bit better which pulse needs to be readout for the SideHcal in this [commit](https://github.com/LDMX-Software/Hcal/pull/5/commits/6904b8328229497a1a88575d1c78fa3c4acef623). 

In particular here:
```
      // Determine which pulse to digitize
      // For top,left we digitize the positive end (0)
      // For bottom,right we digitize the negative end (1)
      bool is_close = false;
      std::vector<ldmx::HgcrocDigiCollection::Sample> digiToAdd;
      if ((section == ldmx::HcalID::HcalSection::TOP) ||
          (section == ldmx::HcalID::HcalSection::LEFT)) {
        if (end_close == 0) is_close = true;
      } else if ((section == ldmx::HcalID::HcalSection::BOTTOM) ||
                 (section == ldmx::HcalID::HcalSection::RIGHT)) {
        if (end_close == 1) is_close = true;
      }
```
we readout the positive pulse (end=0) for top,left and negative pulse (end=1) for bottom,right.